File: docs/examples/py/src/lifecycle_policy_create.py
import asyncio

from reduct import Client, LifecycleSettings, LifecycleType


async def main():
    async with Client("http://127.0.0.1:8383", api_token="my-token") as client:
        await client.create_bucket("my-bucket", exist_ok=True)

        settings = LifecycleSettings(
            type=LifecycleType.DELETE,
            bucket="my-bucket",
            entries=["py-example"],
            max_age="30d",
            interval="1h",
            when={"&anomaly": {"$eq": 1}},
        )
        await client.create_lifecycle("my-lifecycle", settings)


asyncio.run(main())

File: docs/examples/js/src/lifecycle_policy_create.mjs
import { Client, LifecycleType } from "reduct-js";

const client = new Client("http://127.0.0.1:8383", { apiToken: "my-token" });
await client.getOrCreateBucket("my-bucket");

const settings = {
  lifecycleType: LifecycleType.DELETE,
  bucket: "my-bucket",
  entries: ["js-example"],
  maxAge: "30d",
  interval: "1h",
  when: { "&anomaly": { $eq: 1 } },
};

await client.createLifecycle("my-lifecycle", settings);
